azkaban-common/src/test/java/azkaban/trigger/BasicTimeCheckerTest.java

```diff
@@ -77,6 +76,221 @@ public void basicTimerTest() {
     }
 
     assertTrue(cond.isMet());
+  }
+
+  /**
+   * Test Base Cron Functionality.
+   */
+  @Test
+  public void cronTimerTestCurrentZone() {
```

Comment:
Well. Code maintainability is equally important. By the looks of the code, all 4 - 5 methods do share very similar logic which is duplicated across the tests. (Correct me if I am wrong here) Future changes in code logic would have to result in changes in all the tests individually which is not convenient.

As regards to readability, I think a smaller class with smaller methods would help in favor of large test methods which are replicated across the entire class. This will get worse as we add more tests.